cmd/minikube/cmd/start.go

```diff
@@ -1238,12 +1239,14 @@ func validateCPUCount(drvName string) {
 		availableCPUs = ci
 	}
 
-	if availableCPUs < 2 {
-		if drvName == oci.Docker && runtime.GOOS == "darwin" {
+	switch {
+	case availableCPUs < 2:
+		switch {
+		case drvName == oci.Docker && runtime.GOOS == "darwin":
 			exitIfNotForced(reason.RsrcInsufficientDarwinDockerCores, "Docker Desktop has less than 2 CPUs configured, but Kubernetes requires at least 2 to be available")
-		} else if drvName == oci.Docker && runtime.GOOS == "windows" {
+		case drvName == oci.Docker && runtime.GOOS == "windows":
 			exitIfNotForced(reason.RsrcInsufficientWindowsDockerCores, "Docker Desktop has less than 2 CPUs configured, but Kubernetes requires at least 2 to be available")
```

Comment:
Why do we need 2 different exit code and messages for docker on mac or windows?

Also not clear why we need special case for docker desktop.   We spend too much code and effort on this warning. Using one case for all is better.